OPHD/UI/Core/RadioButtonGroup.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include "TextControl.h"
+#include "Label.h"
+#include "../../Cache.h"
+#include "../../Constants.h"
+
+#include <NAS2D/Signal/Signal.h>
+#include <NAS2D/Signal/Delegate.h>
+#include <NAS2D/EventHandler.h>
+#include <NAS2D/Resource/Image.h>
+#include <NAS2D/Utility.h>
+#include <NAS2D/Renderer/Renderer.h>
+#include <NAS2D/MathUtils.h>
+
+#include <algorithm>
+#include <string>
+
+class RadioButtonGroup : public Control
+{
+public:
+	struct ButtonInfo
+	{
+		std::string name;
+		NAS2D::DelegateX<void> delegate;
+	};
+
+	RadioButtonGroup() = default;
+	RadioButtonGroup(std::vector<ButtonInfo> buttonInfos);
+	void setSelected(std::size_t index);
+	void onMove(NAS2D::Vector<int> displacement) override;
+	void update() override;
+	void clearSelection();
+
+private:
+    class RadioButton : public TextControl
+    {
+    public:
+        RadioButton(RadioButtonGroup* parentContainer, std::string newText, NAS2D::DelegateX<void> delegate);
+        ~RadioButton() override;
+        void checked(bool toggle);
+        bool checked() const;
+        void text(const std::string& text);
+        const std::string& text() const;
+        void update() override;
+
+    protected:
+        void onResize() override;
+        void onTextChange() override;
+        void parentContainer(RadioButtonGroup* parent);
+        void onMouseDown(NAS2D::EventHandler::MouseButton button, int x, int y);
+
+    private:
+        const NAS2D::Font& mFont;
+        const NAS2D::Image& mSkin;
+        Label mLabel;
+        RadioButtonGroup* mParentContainer{nullptr};
+        bool mChecked{false};
+        NAS2D::DelegateX<void> mRbgDelegate;
```

Comment:
We should probably revisit `DelegateX` versus `Delegate` at some point. I'm still not clear on why we're using `DelegateX` in this one class.

Reply:
This is still a rather tough one for me, does @ldicker83 perhaps know how we would have to approach this? As we did try some configurations but the compiler would have none of it.

Reply:
Ahh, because most code uses `Signal` rather than `Delegate`. The template parameter list has similar styles for `Signal` and `DelegateX`, but a different style for `Delegate`.

Actually, this file is a bit odd since `mSignal` is defined as a `Delegate`, rather than as a `Signal` object. That makes the API a bit inconsistent here, since `RadioButton` objects can then have only a single listener, while most other controls allow for an arbitrary number of listeners. Of course, we removed the accessor for the `mSignal` object, so there is no way to attach more listeners anyway.

Granted, it makes sense to supply a single listener when constructing the `RadioButton`, so I'm fine with using `Delegate`/`DelegateX` for the `RadioButtonInfo` struct and the `RadioButton` constructor.

I'm wondering if we should perhaps switch back to using a `Signal` object for `mSignal` internally. I could go either way on adding back the accessor for the `SignalSource` slice of that object. I suppose it would make the API more consistent if we added it back.

----

Ok, so if we want to use `Delegate` instead of `DelegateX`, we would specify a template parameter of `void()` rather than `void`.

If we updated `mSignal` to be a `Signal`, then in the `RadioButton` constructor, we would need to change the assignment to a `connect` call:
```cpp
mSignal.connect(delegate);
```

Reply:
Actually, if we could make use of `Signal<>::DelegateType` to define the type. Or event `decltype(mSignal)::DelegateType` (that might required `typename` in front). That way the delegate declaration would be automatically updated in case the `mSignal` declaration was ever changed.